libsession/src/main/java/org/session/libsession/snode/OnionRequestAPI.kt

```diff
@@ -190,8 +191,19 @@ object OnionRequestAPI {
                 if (unusedSnodes.count() < pathSnodeCount) { throw InsufficientSnodesException() }
                 // Don't test path snodes as this would reveal the user's IP to them
                 guardSnodes.minus(reusableGuardSnodes).map { guardSnode ->
-                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).map {
-                        val pathSnode = unusedSnodes.getRandomElement()
+                    val result = listOf( guardSnode ) + (0 until (pathSize - 1)).mapIndexed() { index, _ ->
+                        var pathSnode = unusedSnodes.getRandomElement()
+
+                        // we want to make sure the last node in the path is above version 2.8.0
+                        // to help with an issue that will disappear once the nodes are all updated
+                        if(index == pathSize - 2) {
```

Comment:
Why `pathsize - 2`?

Reply:
`until` is non inclusive. This is because the first node is actually captured with `guardSnode` 
` listOf( guardSnode ) +`

Reply:
Okey doke!